packages/wallet-ui/src/components/ui/molecule/AccountSwitchModal/AccountSwitchModal.view.tsx

```diff
@@ -17,79 +19,199 @@ import { theme } from 'theme/default';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { useAppSelector } from 'hooks/redux';
 import { useStarkNetSnap } from 'services';
+import IconButton from '@mui/material/IconButton';
+import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
+import VisibilityIcon from '@mui/icons-material/Visibility';
+import { Box } from '@mui/material';
 
 interface Props {
   currentAddress: string;
-  accounts: string[];
-  accountsIndex: number[];
   full?: boolean;
   starkName?: string;
 }
 
 export const AccountSwitchModalView = ({
   currentAddress,
-  accounts,
-  accountsIndex,
   full,
   starkName,
 }: Props) => {
   const networks = useAppSelector((state) => state.networks);
-  const { switchAccount, addNewAccount } = useStarkNetSnap();
+  const accounts = useAppSelector((state) => state.wallet.accounts);
+  const { switchAccount, addNewAccount, hideAccount, unHideAccount } =
+    useStarkNetSnap();
   const chainId = networks?.items[networks.activeNetwork]?.chainId;
 
+  const [showHiddenAccounts, setShowHiddenAccounts] = useState(false);
+
+  useEffect(() => {
+    console.log('Updated accounts:', accounts);
+  }, [accounts]);
+
+  const visibleAccounts = accounts.filter(
+    (account) => account.visibility !== false,
+  );
+  const hiddenAccounts = accounts.filter(
+    (account) => account.visibility === false,
+  );
```

Comment:
```suggestion
   const visibleAccounts:Account[] = []
   const hiddenAccounts:Account[] = []
   for (const account in accounts) {
       // account.visibility = `undefined` refer to the case when previous account state doesnt include this field
       // hence we consider it is `visible`
       if (account.visibility === undefined || account.visibility === true) {
          visibleAccounts.append(account)
       } else {
          hiddenAccounts.append(account)
       }
   }
```
1 pass and state more clear on undefined case